test/lokassert.hpp

```diff
@@ -115,16 +115,15 @@ inline constexpr bool failed() { return false; }
         using namespace test::detail;                                                              \
         if (!failed())                                                                             \
         {                                                                                          \
-            auto&& cond##__LINE__ = !!(condition);                                                 \
-            if (!cond##__LINE__)                                                                   \
+            if (!(condition))                                                                      \
             {                                                                                      \
                 std::ostringstream oss##__LINE__;                                                  \
                 oss##__LINE__ << message;                                                          \
                 const auto msg##__LINE__ = oss##__LINE__.str();                                    \
                 TST_LOG("ERROR: Assertion failure: "                                               \
                         << (msg##__LINE__.empty() ? "" : msg##__LINE__ + ". ")                     \
                         << "Condition: " << (#condition));                                         \
-                LOK_ASSERT_IMPL(cond##__LINE__);                                                   \
+                LOK_ASSERT_IMPL(!#condition); /* NOLINT(misc-static-assert) */                     \
```

Comment:
Aha, we know that the condition is false here, so this is assert(false) with a better output.

And then clang-tidy would lint about this, but we silence it, similar to how common/Log.hpp does it.

OK, makes sense now.